framework/platform/bpl/uci/cfg/bpl_cfg_helper.cpp

```diff
@@ -82,14 +82,17 @@ int cfg_get_prplmesh_param_int(const std::string &param, int *buf)
     return RETURN_OK;
 }
 
-int cfg_get_channel(int index, int *channel)
+int cfg_get_channel(const std::string &interface_name, int *channel)
 {
     if (!channel) {
         return RETURN_ERR;
     }
 
     char channel_num[MAX_UCI_BUF_LEN] = {0};
-    int status = cfg_uci_get_wireless(TYPE_RADIO, index, "channel", channel_num);
+    char ifname[BPL_IFNAME_LEN]       = {0};
+
+    utils::copy_string(ifname, interface_name.c_str(), BPL_IFNAME_LEN);
+    int status = cfg_uci_get_wireless_from_ifname(TYPE_RADIO, ifname, "channel", channel_num);
     if (status == RETURN_ERR)
```

Comment:
This is silly - just change it to;
```
if (cfg_uci_get_wireless_from_ifname(TYPE_RADIO, ifname, "channel", channel_num) != RETURN_OK) {
    return RETURN_ERR;
}
```

Reply:
I agree - already fixed it locally - will push the commit after fixing all the comments